cmd/ntpcheck/cmd/utils.go

```diff
@@ -161,7 +161,12 @@ func ntpDate(remoteServerAddr string, remoteServerPort string, requests int) err
 		log.Debugf("Server TX timestamp: %v", serverTransmitTime)
 		log.Debugf("Client RX timestamp: %v", clientReceiveTime)
 
-		avgNetworkDelay := ntp.AvgNetworkDelay(clientTransmitTime, serverReceiveTime, serverTransmitTime, clientReceiveTime)
+		// sanity check: origin time must be same as our transmit time
+		if response.OrigTimeSec != sec || response.OrigTimeFrac != frac {
```

Comment:
Couldn't you just compare `clientTransmitTime` to `clientOriginTime`?

Reply:
No, as if we do it we'll have to deal with rounding errors during the conversion between `time.Time` and NTP time.